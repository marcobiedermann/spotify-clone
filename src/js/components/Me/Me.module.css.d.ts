declare const styles: {
  readonly me: string;
  readonly me__image: string;
};
export = styles;
