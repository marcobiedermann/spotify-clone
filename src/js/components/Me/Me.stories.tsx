import type { Meta, StoryObj } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import Me from '.';
import meFixtures from './__fixtures__';

const meta: Meta<typeof Me> = {
  component: Me,
  args: meFixtures,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof Me>;

const Default: Story = {};

export { Default };
export default meta;
