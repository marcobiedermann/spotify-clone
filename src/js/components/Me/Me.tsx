import clsx from 'clsx';
import React from 'react';
import { Link } from 'react-router-dom';
import Image from '../Image';
import styles from './Me.module.css';

interface Image {
  height: number;
  url: string;
  width: number;
}

export interface MeProps {
  className?: string;
  display_name: string;
  id: string;
  images: Image[];
}

function Me(props: MeProps): JSX.Element {
  const { className, display_name, id, images } = props;

  return (
    <div className={clsx(className, styles.me)}>
      <figure className={styles.me__image}>
        {images[0] && <Image url={images[0].url} alt={display_name} width={32} height={32} />}
      </figure>
      <Link to={`/users/${id}`}>{display_name}</Link>
    </div>
  );
}

export default Me;
