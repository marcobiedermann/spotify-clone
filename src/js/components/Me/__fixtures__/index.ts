import { MeProps } from '..';

const me: MeProps = {
  display_name: 'string',
  id: 'string',
  images: [
    {
      url: 'https://i.scdn.co/image/ab67616d00001e02ff9ca10b55ce82ae553c8228',
      height: 300,
      width: 300,
    },
  ],
};

export default me;
