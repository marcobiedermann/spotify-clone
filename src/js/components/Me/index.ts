import Me from './Me';

export default Me;
export * from './Me';
