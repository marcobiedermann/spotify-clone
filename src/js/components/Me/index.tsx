import classNames from 'classnames';
import React from 'react';
import { Link } from 'react-router-dom';
import Image from '../Image';
import styles from './style.module.css';

interface Image {
  height: number;
  url: string;
  width: number;
}

export interface MeProps {
  className?: string;
  display_name: string;
  id: string;
  images: Image[];
}

function Me(props: MeProps): JSX.Element {
  const { className, display_name, id, images } = props;

  return (
    <div className={classNames(className, styles.me)}>
      <figure className={styles.me__image}>
        {images[0] && <Image url={images[0].url} alt={display_name} width={32} height={32} />}
      </figure>
      <Link to={`/users/${id}`}>{display_name}</Link>
    </div>
  );
}

export default Me;
