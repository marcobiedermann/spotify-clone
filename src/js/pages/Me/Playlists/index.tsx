import React from 'react';
import { Helmet } from 'react-helmet';
import { RouteChildrenProps } from 'react-router-dom';

export type PlaylistsPageProps = RouteChildrenProps;

function PlaylistsPage(): JSX.Element {
  return (
    <>
      <Helmet>
        <title>Playlists</title>
      </Helmet>
      Playlists
    </>
  );
}

export default PlaylistsPage;
