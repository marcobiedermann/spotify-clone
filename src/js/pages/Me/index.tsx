import React from 'react';
import { Helmet } from 'react-helmet';
import { RouteChildrenProps } from 'react-router-dom';

export type MePageProps = RouteChildrenProps;

function MePage(): JSX.Element {
  return (
    <>
      <Helmet>
        <title>Me</title>
      </Helmet>
      Me
    </>
  );
}

export default MePage;
