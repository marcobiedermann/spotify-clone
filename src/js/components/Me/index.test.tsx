import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Me from '.';
import meFixtures from './__fixtures__';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Me {...meFixtures} />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
