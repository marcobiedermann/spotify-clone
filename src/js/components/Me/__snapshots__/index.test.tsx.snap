// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="me"
>
  <figure
    className="me__image"
  >
    <img
      alt=""
      height={32}
      src=""
      width={32}
    />
  </figure>
  <a
    href="/users/"
    onClick={[Function]}
  >
    
  </a>
</div>
`;
